pkgs/tools/networking/vpn-slice/default.nix

```diff
@@ -1,20 +1,26 @@
-{ lib, buildPythonApplication, python3Packages, fetchFromGitHub }:
+{ lib, buildPythonApplication, nix-update-script, python3Packages, fetchFromGitHub }:
 
 buildPythonApplication rec {
   pname = "vpn-slice";
-  version = "0.14";
+  version = "0.15";
 
   src = fetchFromGitHub {
     owner = "dlenski";
     repo = pname;
     rev = "v${version}";
-    sha256 = "1z2mdl3arzl95zrj4ir57f762gcimmmq5nk91j679cshxz4snxyr";
+    sha256 = "sha256-9JnRuJkpcgW1cEgJPiqFDYotNSCl7XcmbHS6D4E13gM=";
   };
 
   propagatedBuildInputs = with python3Packages; [ setproctitle dnspython ];
 
   doCheck = false;
 
+  passthru = {
+    updateScript = nix-update-script {
+      attrPath = pname;
+    };
+  };
+
```

Comment:
I guess, r-ryantm is able to detect new release from GitHub even if the package has no update script thanks to GitHub API. A random example is #144555, if you run this locally:

```
$ nix-shell maintainers/scripts/update.nix --argstr package py3c
```

It will tell you:

```
error: Package with an attribute name `py3c` does not have a `passthru.updateScript` attribute defined.
```

But still r-ryantm is able to open a PR for this. Sorry I don't know Haskell and maybe you can check maybe [here](https://github.com/ryantm/nixpkgs-update/blob/master/src/GH.hs) for how this is done.


And of course if you set a update script for a package, r-ryantm will always run the update script directly instead of figuring out the latest version itself, for example #135202, you will see what "Updates performed" says is different from #144555 ("Ran passthru.UpdateScript" instead of "Version update") although both package fetch their source from GitHub.

And `nix-update-script` is actually calling [nix-update](https://github.com/Mic92/nix-update):

https://github.com/NixOS/nixpkgs/blob/5d7cfc43bba88505c5bba0650dc9f1f4bf69c1b6/pkgs/top-level/all-packages.nix#L112

https://github.com/NixOS/nixpkgs/blob/5d7cfc43bba88505c5bba0650dc9f1f4bf69c1b6/pkgs/common-updater/nix-update.nix#L1-L5

And nix-update itself also supports figure out the latest version of packages from GitHub. I guess keeping the `passthru.updateScript = nix-update-script { attrPath = pname; };` line should be okay, as it will still be useful for people who get used to using `maintainers/scripts/update.nix`.

Reply:
> I don't see any other packages using nix-update in passthru.updateScript.

because it does not require any changes to the package. Just call ``nix-update vpn-slice``.